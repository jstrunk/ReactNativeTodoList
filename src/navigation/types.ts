import { StackNavigationProp } from "@react-navigation/stack";
import { RouteProp } from "@react-navigation/native";

export type RootStackParamList = {
  TodoSelect: undefined;
  TodoList: {id: string};
};

export type TodoSelectScreenNavigationProp = StackNavigationProp<
  RootStackParamList,
  'TodoSelect'
>;

export type TodoListScreenNavigationProp = StackNavigationProp<
  RootStackParamList,
  'TodoList'
>;

export type TodoListScreenRouteProp = RouteProp<
  RootStackParamList,
  'TodoList'
>;

