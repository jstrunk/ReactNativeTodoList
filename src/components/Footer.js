import React, {Component} from 'react';
import {Text,StyleSheet,TouchableOpacity} from 'react-native';

const styles = StyleSheet.create({
  footer: {
    backgroundColor: 'white',
    padding: 15,
    position: 'absolute',
    left: 0,
    right: 0,
    bottom: 0,
  },
  remove: {
    textAlign: 'center',
    color: 'red',
  },
});

export default class Footer extends Component {

  _onPressButton() {
    let items = [];
    this.props.items.map((item) => {
      if (!item.done) items.push(item);
    });
    this.props.onUpdate(items);
  }

  render() {
    return (
      <TouchableOpacity style={styles.footer} onPress={this._onPressButton.bind(this)}>
        <Text style={styles.remove}>{this.props.children}</Text>
      </TouchableOpacity>
    );
  }
}