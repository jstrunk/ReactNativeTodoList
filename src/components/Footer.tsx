import React, {Component} from 'react';
import {Text} from 'react-native';
import DeleteCompleted from '../containers/DeleteCompleted';
import styles from '../styles';

interface Props {
  children: string;
}

export default class Footer extends Component<Props> {
  render(): JSX.Element {
    return (
      <DeleteCompleted style={styles.footer}>
        <Text style={styles.remove}>{this.props.children}</Text>
      </DeleteCompleted>
    );
  }
}
