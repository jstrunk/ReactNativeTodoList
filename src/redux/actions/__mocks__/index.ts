export const addItem = jest.fn();
export const toggleItem = jest.fn();
export const deleteItem = jest.fn(); 
export const deleteCompleted = jest.fn(); 
export const reorderList = jest.fn(); 
export const registerList = jest.fn(); 
export const setActiveList = jest.fn(); 
export const createList = jest.fn(); 
export const loadList = jest.fn(); 