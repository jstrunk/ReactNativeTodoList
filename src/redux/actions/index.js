import { ADD_ITEM, DELETE_ITEM, TOGGLE_ITEM, DELETE_COMPLETED } from '../constants';

export function addItem(text) {
  return { type: ADD_ITEM, text };
}

export function toggleItem(index) {
  return { type: TOGGLE_ITEM, index };
}

export function deleteItem(index) {
  return { type: DELETE_ITEM, index };
}

export function deleteCompleted() {
  return { type: DELETE_COMPLETED };
}
