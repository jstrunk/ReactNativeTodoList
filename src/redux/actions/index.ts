import { ADD_ITEM,
  DELETE_ITEM,
  TOGGLE_ITEM,
  DELETE_COMPLETED,
  TodoActionType, } from '../types';

export function addItem(text: string): TodoActionType {
  return { type: ADD_ITEM, text };
}

export function toggleItem(index: number): TodoActionType {
  return { type: TOGGLE_ITEM, index };
}

export function deleteItem(index: number): TodoActionType {
  return { type: DELETE_ITEM, index };
}

export function deleteCompleted(): TodoActionType {
  return { type: DELETE_COMPLETED };
}
