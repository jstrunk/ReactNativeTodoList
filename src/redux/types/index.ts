export interface Todo {
  key: string;
  done: boolean;
}

export interface ITodoList {
  byId: Record<string,Todo>;
  allItems: Array<string>;
}

export interface ITodoState {
  todoList: ITodoList;
}

export const defaultState: ITodoList = {
  byId: {},
  allItems: [],
};

export const ADD_ITEM = "ADD_ITEM";
export const TOGGLE_ITEM = "TOGGLE_ITEM";
export const DELETE_ITEM = "DELETE_ITEM";
export const DELETE_COMPLETED = "DELETE_COMPLETED";

interface AddItemAction {
  type: typeof ADD_ITEM;
  text: string;
  id: string;
}

interface ToggleItemAction {
  type: typeof TOGGLE_ITEM;
  id: string;
}

interface DeleteItemAction {
  type: typeof DELETE_ITEM;
  id: string;
}

interface DeleteCompletedAction {
  type: typeof DELETE_COMPLETED;
  ids: Array<string>;
}

export type TodoActionType = AddItemAction | ToggleItemAction |
  DeleteItemAction | DeleteCompletedAction;
