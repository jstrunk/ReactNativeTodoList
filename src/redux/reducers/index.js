import { combineReducers } from 'redux';
import listReducer from '../reducers/listReducer';

export default combineReducers(
  { todoList: listReducer }
);