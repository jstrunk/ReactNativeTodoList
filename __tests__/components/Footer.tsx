import 'react-native';
import React from 'react';
import Footer from '../../src/components/Footer';
import * as types from '../../src/redux/types';
import * as actions from '../../src/redux/actions';
import { Provider } from 'react-redux';
import { render, fireEvent } from '@testing-library/react-native'
import configureMockStore, { MockStore } from 'redux-mock-store';
import thunk, { ThunkDispatch } from 'redux-thunk';
import { oneListOneItemDone } from '../../test/testStates';

jest.mock('../../src/redux/actions');

type DispatchExts = ThunkDispatch<types.ITodoState, void, types.CombinedActionType>;
const middlewares = [thunk];
const mockStore = configureMockStore<types.ITodoState,DispatchExts>(middlewares);

describe('Footer component', () => {
  let store: MockStore;

  beforeEach(() => {
    store = mockStore(oneListOneItemDone);

    store.dispatch = jest.fn();
  });

  it('renders correctly', () => {
    const { toJSON } = render(
      <Provider store={store}>
        <Footer>Delete Completed</Footer>
      </Provider>
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should dispatch a delete completed items action', () => {
    const { getByText } = render(
      <Provider store={store}>
        <Footer>Delete Completed</Footer>
      </Provider>
    );

    const element = getByText('Delete Completed');
    fireEvent.press(element);

    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledWith(
      actions.deleteCompleted(['abc1'])
    );
    expect(actions.deleteCompleted).toHaveBeenCalledWith(['abc1']);
  });
});