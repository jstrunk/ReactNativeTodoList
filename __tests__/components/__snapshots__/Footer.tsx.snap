// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "bottom": 0,
      "left": 0,
      "padding": 15,
      "position": "absolute",
      "right": 0,
    }
  }
>
  <Text
    style={
      Object {
        "color": "red",
        "textAlign": "center",
      }
    }
  >
    Delete Completed
  </Text>
</View>
`;
